examples/src/bin/ech-client.rs

```diff
@@ -201,37 +215,50 @@ struct Args {
     inner_hostname: String,
 }
 
+/// Collect up all `EchConfigListBytes` found in the HTTPS record(s) for a given domain name/port.
+///
+/// The domain name should be the **inner** name used for Encrypted Client Hello (ECH). The
+/// lookup is done using DNS-over-HTTPS to protect that inner name from being disclosed in
+/// plaintext ahead of the TLS handshake that negotiates ECH for the inner name.
+///
+/// Returns an empty vec if no HTTPS records with ECH configs are found.
 // TODO(@cpu): consider upstreaming to hickory-dns
 async fn lookup_ech_configs(
     resolver: &TokioResolver,
     domain: &str,
     port: u16,
-) -> Option<EchConfigListBytes<'static>> {
-    // For non-standard ports, lookup the ECHConfig using port-prefix naming
-    // See: https://datatracker.ietf.org/doc/html/rfc9460#section-9.1
+) -> Vec<EchConfigListBytes<'static>> {
     let qname_to_lookup = match port {
         443 => domain.to_owned(),
         port => format!("_{port}._https.{domain}"),
     };
 
-    resolver
+    let Ok(lookup) = resolver
```

Comment:
Hmm, should we be silently discarding the error here?

Reply:
Yeah I think it'd be better to return the error. Updated.